### ISSUE TITLE
[BUG] lammps_scatter_atoms_subset method (in library.cpp) doesn't work properly

### ISSUE THREAD

**Issue by erozic** [lammps/lammps#1343]
**Summary**

The ```lammps_scatter_atoms_subset``` method of ```library.cpp``` doesn't update the atoms given by the ```ıds``` array, but sequentially (as in ```lammps_scatter_atoms```), when the data type is ```int```.

This is most probably just a c/p mistake from the aforementioned (older) ```lammps_scatter_atoms``` method, and the correct code can be seen a few lines lower in the ```double``` data type case, namely:

```
if (count == 1) {
  for (i = 0; i < ndata; i++) {
    id = ids[i];
    if ((m = lmp->atom->map(id)) >= 0)
      vector[m] = dptr[i];
  }
} else ...
```

**LAMMPS Version and Platform**

In the current master branch and the latest stable release (12Dec2018)

**Expected Behavior**

As described in the method description in ```library.cpp```...

**Actual Behavior**

It doesn't update the atoms specified by ```ids``` but the first ```ndata``` on the local proc.

**Steps to Reproduce**

Just call the method on whatever single-value integer property, like "type" or "id"...

**Comment by erozic** [comment]
By the way, can you tell me is this the preferred/correct way to report this or should I have just corrected this in my fork and submitted a pull request in which I explain what I did and why? Since it's obvious what the problem is and I have fixed it and tested it locally for myself...

**Comment by akohlmey** [comment]
> By the way, can you tell me is this the preferred/correct way to report this or should I have just corrected this in my fork and submitted a pull request in which I explain what I did and why? Since it's obvious what the problem is and I have fixed it and tested it locally for myself...

@erozic your bug report is fine like this, but if you have a working change, a pull request would be preferred. It will most certainly improve the chance, that a bugfix is included. On github, a pull request is essentially an issue with an associated branch. If you submit a pull request, a bug report is not really necessary, but since you have already created one, you can just add the text `closes #1343` to the description of the pull request (under **Related Issues**) and then the issue will be automatically closed, when the pull request is approved and merged.

**Comment by erozic** [comment]
Great, thanks, doing that now :)
